x/vm/types/scaling.go

```diff
@@ -9,23 +9,6 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// MustConvertEvmCoinTo18Decimals converts the coin's Amount from its original
-// representation into a 18 decimals. The function panics if coin denom is
-// not the evm denom or in case of overflow.
-//
-// CONTRACT: The function should only be called when the coin denom is the EVM. This means that
-// should be called only when the code forces the denom to be the expected one.
-func MustConvertEvmCoinTo18Decimals(coin sdk.Coin) sdk.Coin {
-	if coin.Denom != GetEVMCoinDenom() {
-		panic(fmt.Sprintf("expected evm denom %s, received %s", GetEVMCoinDenom(), coin.Denom))
-	}
-
-	evmCoinDecimal := GetEVMCoinDecimals()
-	newAmount := coin.Amount.Mul(evmCoinDecimal.ConversionFactor())
-
-	return sdk.Coin{Denom: coin.Denom, Amount: newAmount}
-}
-
 // ConvertAmountToLegacy18Decimals convert the given amount into a 18 decimals
 // representation.
 func ConvertAmountTo18DecimalsLegacy(amt sdkmath.LegacyDec) sdkmath.LegacyDec {
```

Comment:
How about adding unit tests for `ConvertAmountTo18DecimalsLegacy` and `ConvertAmountTo18DecimalsBigInt`, although this is indirectly covered in the other tests, it’s still a core function and internal logic changed by `ConversionFactor`

Reply:
I added tests for those util funcs.
ref. https://github.com/cosmos/evm/pull/69/commits/9516f672b609b39a32236b7ae1148f04cca8c48a